packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import { StyledActionListItem } from './StyledActionListItem';
+import { componentIds } from './componentIds';
+import type { StyledActionListItemProps } from './getBaseActionListItemStyles';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { isReactNative, makeAccessible, makeSize } from '~utils';
+import { Checkbox } from '~components/Checkbox';
+
+const ActionListItemContext = React.createContext<{
+  intent?: ActionListItemProps['intent'];
+}>({});
+
+const SectionDivider = styled(Box)((props) => ({
+  // @TODO: replace this with token value if we add 1px token
+  height: makeSize(1),
+  backgroundColor: props.theme.colors.surface.border.normal.lowContrast,
+  margin: `${makeSize(props.theme.spacing[1])} ${makeSize(props.theme.spacing[3])}`,
+}));
+
+const StyledActionListSectionTitle = styled(Box)((props) => ({
+  padding: makeSize(props.theme.spacing[3]),
+}));
+
+const ActionListSection = ({
+  title,
+  children,
+  hideDivider,
+}: {
+  title: string;
+  children: React.ReactNode[] | React.ReactNode;
+  hideDivider?: boolean;
+}): JSX.Element => {
+  return (
+    <Box>
+      <StyledActionListSectionTitle>
+        <Text color="surface.text.muted.lowContrast" size="small" weight="bold">
+          {title}
+        </Text>
+      </StyledActionListSectionTitle>
+      {children}
+      {hideDivider ? null : <SectionDivider />}
+    </Box>
+  );
+};
+
+ActionListSection.componentId = componentIds.ActionListSection;
+
+const ActionListItemIcon = ({ icon }: { icon: IconComponent }): JSX.Element => {
+  const Icon = icon;
+  const { intent } = React.useContext(ActionListItemContext);
+  return (
+    <Icon
+      color={
+        intent === 'negative'
+          ? 'feedback.icon.negative.lowContrast'
+          : 'surface.text.muted.lowContrast'
+      }
+      size="medium"
+    />
+  );
+};
+
+const ActionListItemText = ({ children }: { children: string }): JSX.Element => {
+  return (
+    <Text variant="caption" color="surface.text.muted.lowContrast">
+      {children}
+    </Text>
+  );
+};
+
+const ActionListCheckboxWrapper = styled(Box)((_props) => ({
+  pointerEvents: 'none',
+}));
+
+const getActionListItemRole = (href?: string): 'link' | 'menuitem' | 'option' => {
+  if (href) {
+    return 'link';
+  }
+
+  if (isReactNative()) {
+    return 'menuitem';
+  }
+
+  return 'option';
+};
+
+/**
+ *
+ * ActionListItem
+ *
+ *
+ */
+type ActionListItemProps = {
+  title: string;
+  description?: string;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
is passing the entire event necessary? or just few args like name and value as we do in input fields will suffice?

Reply:
don't see it necessary as of now. Changed it to `{ name: string; value: boolean }` (name is the name of option and value is the `isSelected` boolean)